### ISSUE TITLE
Delay when a running query is cancelled in mysql while using Teleport DB access

### ISSUE THREAD

**Issue by oshati** [gravitational/teleport#13668]
Expected behaviour:
When Teleport is used to access mysql DB, any running query that is cancelled should be immediately aborted and process should stop almost immediately. 

Current behaviour:
When a query command is running and an attempt is made to cancel the query, there is a 10 secs delay before the action to cancel works and despite showing a "query aborted" prompt immediately. 

Bug details:
- Teleport version 9.0.4 (This was also tested on an newer version 9.3.3 and experience was the same)
- Steps to recreate: 
1. Login to the proxy tsh login xxx
2. connect to the specific db tsh db connect xxx
3. Run a sample query e.g. select sleep(10);
After running the query, press ctrl+c to cancel it. You get immediately “query aborted”, but it still takes 10 secs or more to return to complete the cancelation process.

**Comment by pkatiushyn** [comment]
Any update on this? It was fixed for Postgres: https://github.com/gravitational/teleport/pull/22173.

**Comment by smallinsky** [comment]
@pkatiushyn 
No, the issue is still open. The MySQL cancelation mechanism differs from Postgres cancellation approach and is more complex in terms of implementation inside Teleport Database Access flow.
